config/cmake/HDFCompilerFlags.cmake

```diff
@@ -189,6 +189,19 @@ if (HDF5_ENABLE_DEV_WARNINGS)
   elseif (CMAKE_C_COMPILER_ID MATCHES "IntelLLVM" OR CMAKE_C_COMPILER_ID MATCHES "[Cc]lang")
     ADD_H5_FLAGS (H5_CFLAGS "${HDF5_SOURCE_DIR}/config/clang-warnings/developer-general")
   endif ()
+
+  # Turn on -Winline warnings now only for non-Debug and
+  # non-Developer builds. For at least GNU compilers this
+  # flag appears to conflict specifically with the -Og
+  # optimization flag and will produce warnings about functions
+  # not being considered for inlining
+  if (NOT ${HDF_CFG_NAME} MATCHES "Debug" AND NOT ${HDF_CFG_NAME} MATCHES "Developer")
```

Comment:
I have no problem with this arrangement.

Does release_docs/INSTALL_Warnings.txt section II. Default Warnings need changed?

Reply:
Looks like yes, but to be honest this looks like just another file that adds maintenance burden rather than directing people to the relevant files where they can see which warnings get added in

Reply:
Agree, just another issue to add to the list of documentation fixes.